biojava-structure/src/main/java/org/biojava/nbio/structure/cluster/SubunitClusterer.java

```diff
@@ -74,14 +74,10 @@ public static Stoichiometry cluster(List<Subunit> subunits, SubunitClustererPara
 						if (params.isUseEntityIdForSeqIdentityDetermination() &&
 								clusters.get(c1).mergeIdenticalByEntityId(clusters.get(c2))) {
 							// This we will only do if the switch is for entity id comparison is on.
-							// In some cases in can save enormous amounts of time, e.g. for clustering full
+							// In some cases it can save enormous amounts of time, e.g. for clustering full
 							// chains of deposited PDB entries. For instance for 6NHJ: with pure alignments it
 							// takes ~ 6 hours, with entity id comparisons it takes 2 minutes.
 							clusters.remove(c2);
-						} else if (clusters.get(c1).mergeIdentical(clusters.get(c2))) {
-							// This always makes sense as an optimization: it's far cheaper to compare the sequence
-							// string than doing a full S-W alignment
```

Comment:
I think you are correct here but there are some problems like missing residues at the ends or modified amino acids, so using S-W alignment was an easy solution at the time, but clearly not optimal. Do you have an idea why the test is failing?

Reply:
With this optimization enabled, the test that fails is `TestQuatSymmetryDetectorExamples.testLocal()`, but I have no idea why. I then decided to remove this optimization as anyway for my use-case the entity id comparison takes care of it.

Reply:
Ok makes sense! Thanks